rflx/pyrflx/typevalue.py

```diff
@@ -189,30 +254,74 @@ def literals(self) -> Mapping[Name, Expr]:
         return {Variable(k): v for k, v in self._type.literals.items()}
 
 
-class OpaqueValue(TypeValue):
+class CompositeValue(TypeValue):
+
+    _expected_size: Optional[Expr]
+
+    def __init__(self, vtype: Composite) -> None:
+        self._expected_size = None
+        super().__init__(vtype)
```

Comment:
I think it is sufficient to declare `_expected_size` in `__init__`.

Reply:
You would have to add the type in `__init__` then. As the attribute is defined in `__init__`, I would remove the type declaration and change the definition in `__init__` to `self._expected_size: Expr = None`. Note that you do not need `Optional` in this case, because mypy can infer that from the initialization with `None`.

Reply:
If i declare it as `self._expected_size: Expr = None` i get a mypy error:
`Incompatible types in assignment (expression has type "None", variable has type "Expr")`.
It works with Optional[Expr]

Reply:
I see, infering `Optional` seems only to work for parameters with a default value `None`.